### ISSUE TITLE
Grant Produces Unparsable Section on Multiple Grants

### ISSUE THREAD

**Issue by mariugul** [sqlfluff/sqlfluff#2016]
## Expected Behaviour
Granting priviliges works on single tables or roles but it does not work when you add multiple. For example granting like the below SQL produces "unparsable" output.

```sql
GRANT SELECT, INSERT, UPDATE, DELETE 
ON my_table1, my_table2, my_table3, my_table4
TO "some-something";
```

While granting like this does not: 

```sql
GRANT SELECT, INSERT, UPDATE, DELETE 
ON my_table1
TO "some-something";
```

Additionally, a table or role name with a hyphen produces an PostgreSQL error, but the linter is not able to detect this error. It's produced as "unparsable".

## Observed Behaviour

```bash
$ sqlfluff lint --dialect postgres grant.sql 
== [grant.sql] FAIL                                                          
L:  12 | P:   1 |  PRS | Line 12, Position 1: Found unparsable section: 'GRANT
                       | INSERT ON my_table, my_table2 TO p...'
L:  13 | P:   1 |  PRS | Line 13, Position 1: Found unparsable section: 'GRANT
                       | INSERT ON my_table, my_table2 TO "...'
L:  14 | P:   1 |  PRS | Line 14, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE ON my_table, my_tab...'
L:  15 | P:   1 |  PRS | Line 15, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE ON my_table, my_tab...'
L:  16 | P:   1 |  PRS | Line 16, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE, DELETE ON my_table...'
L:  17 | P:   1 |  PRS | Line 17, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE, DELETE ON my_table...'
L:  18 | P:   1 |  PRS | Line 18, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE, DELETE, SELECT ON ...'
L:  19 | P:   1 |  PRS | Line 19, Position 1: Found unparsable section: 'GRANT
                       | INSERT, UPDATE, DELETE, SELECT ON ...'
L:  22 | P:   1 |  PRS | Line 22, Position 1: Found unparsable section: 'GRANT
                       | SELECT, INSERT, UPDATE, DELETE ON ...'
All Finished 📜 🎉!
```

## Steps to Reproduce
**grant.sql**:

```sql
-- Works
GRANT INSERT ON my_table TO public;
GRANT INSERT ON my_table TO "public";
GRANT INSERT, UPDATE ON my_table TO public;
GRANT INSERT, UPDATE ON my_table TO "public";
GRANT INSERT, UPDATE, DELETE ON my_table TO public;
GRANT INSERT, UPDATE, DELETE ON my_table TO "public";
GRANT INSERT, UPDATE, DELETE, SELECT ON my_table TO public;
GRANT INSERT, UPDATE, DELETE, SELECT ON my_table TO "public";

-- Doesn't work
GRANT INSERT ON my_table, my_table2 TO public;
GRANT INSERT ON my_table, my_table2 TO "public";
GRANT INSERT, UPDATE ON my_table, my_table2 TO public;
GRANT INSERT, UPDATE ON my_table, my_table2 TO "public";
GRANT INSERT, UPDATE, DELETE ON my_table, my_table2 TO public;
GRANT INSERT, UPDATE, DELETE ON my_table, my_table2 TO "public";
GRANT INSERT, UPDATE, DELETE, SELECT ON my_table, my_table2 TO public;
GRANT INSERT, UPDATE, DELETE, SELECT ON my_table, my_table2 TO "public";

-- Can't parse that the hyphen '-' is illegal
GRANT SELECT, INSERT, UPDATE, DELETE ON my_table1 TO some-something;

```
**Lint grant.sql**:

```bash
sqlfluff lint functional-up.sql --dialect postgres
```

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2`
`Python 3.8.10`
`pip 20.0.2 `

## Configuration
`--dialect postgres`
